### PR TITLE
Update group_by.py

### PR DESCRIPTION
**Pull request by Vebryn**
add note related to when issue

### CODE REVIEW

**Comment by bcoca** [comment]
just a link is not very helpful, also it is not easy to understand the cause from the ticket itself w/o an explanation

**Comment by Vebryn** [comment]
how to merge commits to ansible doc ?

**Comment by gregdek** [comment]
Thanks @Vebryn. @jhoekx please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

**Comment by robynbergeron** [comment]
@jhoekx This change is still pending your review; do you have time to take a look and comment? Please comment with text 'shipit' or 'needs_revision' as appropriate.

File: inventory/group_by.py [hunk 1]
```diff
@@ -30,6 +30,9 @@
 author: "Jeroen Hoekx (@jhoekx)"
 notes:
   - Spaces in group names are converted to dashes '-'.
+  - Use of 'when' directive cause unpredictable results.
```
Comment (jhoekx) [inline inventory/group_by.py]: I'm not a native speaker, but I think the message should be improved.

Something like: 
- Use of group_by with the when statement causes unpredictable results.
- Only variables of the first host in the inventory are available.

**Comment by jhoekx** [comment]
needs_revision

**Comment by gregdek** [comment]
Thanks @Vebryn for this PR. A maintainer of this module has asked for revisions to this PR. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

**Comment by gregdek** [comment]
@Vebryn A friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open pending your changes.

**Comment by Vebryn** [comment]
Action done with commit [310c1b9](https://github.com/ansible/ansible-modules-core/pull/2397/commits/310c1b9f857a922ca92685653b163a96b0121803), any other action needed ?

**Comment by gregdek** [comment]
@Vebryn Another friendly reminder: this pull request has been marked as needing your action. If you still believe that this PR applies, and you intend to address the issues with this PR, just let us know in the PR itself and we will keep it open. If you have addressed the issues and believe it's ready for review, please comment with the text "ready_for_review". If we don't hear from you within another 14 days, we will close this pull request.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by Vebryn** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @Vebryn. To the current maintainers, @jhoekx please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit' or 'needs_revision' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by gregdek** [comment]
Thanks @Vebryn for this PR. This module is maintained by the Ansible core team, so it can take a while for patches to be reviewed. Thanks for your patience.

Core team: please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with 'needs_revision' or merge as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Pull request closed without merging**